### PR TITLE
Excel time to numeric

### LINKED ISSUE #245: Feature Request: excel_numeric_to_time
We currently have `excel_numeric_to_date()` to assist with converting dates and datetimes to POSIXct objects.  Can we create a similar function for `excel_numeric_to_time()` or perhaps more accurately `excel_time_to_numeric()`?

Excel stores times in a format as a fraction of a day, and an issue that I commonly have when opening an Excel file with times is that they are not consistently read correctly (this is also related to tidyverse/readxl#504).

In files that I read, the person providing the file often mixes multiple formats accidentally, so that when read in, I get a mixture like the following:

```r
c("9:04AM", "14:35", "0.395833333333", "1899-12-31 08:05")
```

The issue that I have is usually related to the mixture of formats due to imprecise data entry.  I think it would be a good pairing with the `excel_numeric_to_date()` function to have a function to help handle times.  The interface I was thinking of is below.  The input would be any of the formats above:

* a fraction between 0 (inclusive) and 1 (exclusive) that may be a character string.
* Time with AM/PM (must have AM/PM case insensitively at the end)
* 24-hour time
* A POSIXlt or POSIXct object on the day "1899-12-31"
    * Or a character string that looks like a POSIXlt/POSIXct object from this day, notably the string starts with "1899-12-31 " and then has a 24-hour time.

The return value would be a the number of seconds in the time (a value >=0 and <86400).  Another option for output could be a lubridate duration object (but that would require lubridate as a dependency).

```r
excel_time_to_numeric(time_value, round_seconds=TRUE)
```

If interested, I can make a PR.

### PR DESCRIPTION
**Pull request by billdenney**
Replaces #246 (by switching to the `main` branch) and fixes #245

### CODE REVIEW

**Comment by billdenney** [comment]
@sfirke , Hopefully this can make the next CRAN release.

**Comment by sfirke** [comment]
I can review this this week.

**Comment by billdenney** [comment]
@sfirke, I found myself needing this function today.  So: ping!  😄

**Comment by sfirke** [comment]
Ack sorry I have been so slow!  Thanks for the ping, I will try to review soon.

File: R/excel_time_to_numeric.R
Comment (sfirke) [inline R/excel_time_to_numeric.R:92]: Will it still match midnight with the trailing space after the 1899-12-31?

File: R/excel_time_to_numeric.R
Comment (sfirke) [inline R/excel_time_to_numeric.R:90]:
```suggestion
      # The entire time is optional to allow for midnight which shows as
```

File: R/excel_time_to_numeric.R
Comment (sfirke) [inline R/excel_time_to_numeric.R:166]: Elegant! 👏

File: tests/testthat/test-excel_time_to_numeric.R
Comment (sfirke) [inline tests/testthat/test-excel_time_to_numeric.R:1]: I still need to try out the tests and review them but don't want that to hold up my feedback on the actual code.

**Review by sfirke — COMMENTED** [review]
I looked at the actual code, not the tests yet.  Code looks great, all that thinking about time format regex has me 😵‍💫

File: R/excel_time_to_numeric.R
Comment (sfirke) [reply R/excel_time_to_numeric.R:166]: The function overall, that is.

File: R/excel_time_to_numeric.R
Comment (billdenney) [reply R/excel_time_to_numeric.R:14]: Yes, I fixed that.

File: R/excel_time_to_numeric.R
Comment (billdenney) [reply R/excel_time_to_numeric.R:41]: Good point

File: R/excel_time_to_numeric.R
Comment (billdenney) [reply R/excel_time_to_numeric.R:92]: I added this test which I think confirms that the answer is yes:

```
expect_equal(excel_time_to_numeric(as.POSIXct("1899-12-31 00:01")), 60)
```

**Comment by billdenney** [comment]
I think that I've addressed all the comments.  Can you please take one more look and merge if it looks good to you?

File: R/excel_time_to_numeric.R
Comment (billdenney) [reply R/excel_time_to_numeric.R:166]: Thanks!

File: tests/testthat/test-excel_time_to_numeric.R
Comment (sfirke) [inline tests/testthat/test-excel_time_to_numeric.R:142]: I feel like there must be some story behind this use case 😆

**Review by sfirke — APPROVED** [review]

**Comment by sfirke** [comment]
@billdenney this looks great!  Thank you very much, both for the great code and the patience as I was slow to review.

Feel free to squash and merge.  I thought one block of tests was redundant and wanted to leave the chance for you to remove it (if I'm not missing something) before merging, but this is ready to go.

I wonder if it's a good impetus to start the next CRAN submission, to make this available to a broader user base.

File: tests/testthat/test-excel_time_to_numeric.R
Comment (billdenney) [reply tests/testthat/test-excel_time_to_numeric.R:142]: You'd be amazed at the trash that I have to sort through in some of my source data.  (Or maybe as the originator of `janitor`, you wouldn't 😄 .)  I sometimes get fields that have multiple data points concatenated together.  Dates and times are especially bad.  As long as it fully matches an expected format (maybe with extra bits at the end), I wanted to let it in.

**Pull request merged**
